cos-fleetshard-operator-debezium/src/main/java/org/bf2/cos/fleetshard/operator/debezium/DebeziumOperandSupport.java

```diff
@@ -176,99 +182,160 @@ private static io.fabric8.kubernetes.api.model.Condition cloneCondition(
         return copyiedCondition;
     }
 
-    public static void computeStatus(ConnectorStatusSpec statusSpec, KafkaConnect kafkaConnect, KafkaConnector kafkaConnector) {
-        statusSpec.setConditions(new ArrayList<>());
-
-        statusSpec.setPhase(ManagedConnector.STATE_PROVISIONING);
-
-        var readyCondition = new io.fabric8.kubernetes.api.model.Condition();
-        readyCondition.setType("Ready");
-        readyCondition.setStatus("False");
-        readyCondition.setReason("Transitioning");
-
-        boolean kafkaConnectorFailed = true;
+    private static void computeConnectorStatus(
+        KafkaConnector kafkaConnector,
+        ConnectorStatus connectorStatus) {
 
         if (null != kafkaConnector) {
+            Condition connectorReadyCondition = null;
+            Condition connectorNotReadyCondition = null;
```

Comment:
It is needed to have both ? The Ready and NotReady conditions can be probably collapsed into a single Ready condition where the status is true or false, which could simplify some of the flow control later in this method

Reply:
That's true, but they need to be and are combined later. We never know in which order they are coming in the list of conditions and we need to capture both first while iterating over the list of conditions. Later they are aggregated into one readyCondition. It even gets combined with the KafkaConnect conditions later on.

Reply:
AFAIK, I don't think you can have both Ready and NotReady condition at the same time in fact, there's no Ready/False and NotReady/True (at least last time I checked)

Reply:
@lburgazzoli can you link to where this can be checked?

Reply:
I was looking at this code: https://github.com/strimzi/strimzi-kafka-operator/blob/56342fd3d3f0c86e54e0d44a24474856a2af43e2/operator-common/src/main/java/io/strimzi/operator/common/operator/resource/StatusUtils.java#L92-L104 so my understanding is that the "ready" condition can in fact be either "Ready" or "NotReady".

It is not a common pattern in Kubernetes but it make sense that they are mutually exclusive i.e. can you have a condition for which a resource is both ready and not ready at the same time ?

Maybe it is worth checking with the strimzi dev

Reply:
I expected `{"Ready": "False", "NotReady": "True"}` in some situations. There might even be timing issues where `{"Ready": "True", "NotReady": "True"}` co-exist, even future bugs? When I tried to figure this out I found it very complex to figure out when which condition is set and removed.
The current implementation is resilient to any situation and I would just leave the code as it is and would prefer to not spend even more time on this. I wasted enough time to make the integration tests work.

Reply:
According to the code, the conditions you described should not be possible so I would recommend to get in touch with the strimzi team to have some understanding of what is expected but that's your call.

Reply:
I still lean towards this more robust and resilient implementation here. If we don't see such situations in production for a while I am okay with revisiting this part of the code.